File: 2017/day01-1.ts
function solution(content: string) {
  console.log(
    content
      .split("")
      .map((item, index, array) =>
        item === array[(index + 1) % array.length] ? parseInt(item) : 0
      )
      .reduce((acc, cur) => acc + cur, 0)
  );
}

Deno.readTextFile("day01.dat")
  .then((content) => content.trim())
  .then(solution);

File: 2017/day01-1.hs
import System.IO
import Data.Char

solution :: String -> Int
solution content = foldr (+) 0 . map (\(x,_) -> x) . filter (\(x,y) -> x == y) $ zip parsed ((tail parsed) ++ [head parsed])
    where
        parsed = map digitToInt . trim $ content


trim :: String -> String
trim = filter (\x -> x /= '\n')

main :: IO ()
main = do
    handle <- openFile "day01.dat" ReadMode
    content <- hGetContents handle
    print . solution $ content